02_bubble/n0npax/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Divides slice into partitions base on pivot hight
+func partition(s []int, low int, high int) int {
+	pivot, i := s[high]+1, low-1
+	for j := low; j < high; j++ {
+		if s[j] < pivot {
+			i++
+			s[i], s[j] = s[j], s[i]
+		}
+	}
+	s[i+1], s[high] = s[high], s[i+1]
+	return i + 1
+}
+
+// Sorts provided partition. Modifies reference
+func quicksort(s []int, lo int, hi int) {
+	if lo < hi {
+		// p - partition veretical pivot
+		p := partition(s, lo, hi)
+		quicksort(s, lo, p-1)
+		quicksort(s, p+1, hi)
+	}
+
+}
+
+// Return sorted list(copy) using quicksort
+// Wrapper to unify sort definition
+// Quick sort has average compexity O(nlogn)
+func quicksortWrapper(s []int) []int {
+	ts := copySlice(s)
+
+	quicksort(ts, 0, len(ts)-1)
+	return ts
+}
+
+// Return sorted list(copy) using insertion
+func insertion(s []int) []int {
+	ts := copySlice(s)
+
+	for i := range ts {
+		for i > 0 {
+			if ts[i-1] > ts[i] {
+				ts[i-1], ts[i] = ts[i], ts[i-1]
+			}
+			i--
+		}
+	}
+	return ts
+}
+
+// Return sorted list(copy) using buble sort
+func bubble(s []int) []int {
+	ts := copySlice(s)
+
+	for i1, v1 := range ts {
+		for i2, v2 := range ts[:i1] {
+			if v2 > v1 {
+				ts[i1], ts[i2] = ts[i2], ts[i1]
+			}
+		}
+	}
+	return ts
+}
+
+// Copy slice
+func copySlice(s []int) []int {
```

Comment:
if you insist on a oneline you could do this: `b = append(a[:0:0], a...)` (https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F)

Reply:
fixed. Thanks for hint.

Reply:
I mean I changed function body to one liner. Do you want me to replace all function invocation with provided syntax?